chore: create copy of fs-hdfs

parthchandra opened the pull request:
## Which issue does this PR close?

Part of https://github.com/apache/datafusion-comet/issues/2034

## Rationale for this change

We are using `fs-hdfs` but the project is not responsive and PRs are not getting merged.

## What changes are included in this PR?
This simply adds a copy of the code of `fs-hdfs3` in the Comet code base. The hdfs module code still refers to the published crate(s) of the fs-hdfs project.

## How are these changes tested?

No test yet.

parthchandra commented:
@comphead @Kontinuation @drexler-sky @jiayuasu, fyi
cc @andygrove @mbutrovich

andygrove commented on native/fs-hdfs/Cargo.toml:
```diff
@@ -0,0 +1,55 @@
+# under the License.
+
+[package]
+name = "fs-hdfs3"
```
I suggest that we change the crate name here (similar to what we did for the `hdfs` crate).

```suggestion
name = "datafusion-comet-fs-hdfs3"
```

andygrove commented on native/fs-hdfs/Cargo.toml:
```diff
@@ -0,0 +1,55 @@
+edition = "2021"
+
+description = "libhdfs binding library and safe Rust APIs"
+authors = ["Yanghong Zhong <nju_yaho@apache.org>"]
```
We should probably update this.

```suggestion
authors = ["Apache DataFusion <dev@datafusion.apache.org>"]
```

andygrove commented on native/fs-hdfs/Cargo.toml:
```diff
@@ -0,0 +1,55 @@
+license = "Apache-2.0"
+
+keywords = ["hdfs", "hadoop"]
+documentation = "https://docs.rs/crate/fs-hdfs3"
+homepage = "https://github.com/datafusion-contrib/fs-hdfs"
+repository = "https://github.com/datafusion-contrib/fs-hdfs"
```
We won't be publishing this crate (as far as I know), so we can either remove these entries or update them to point to the Comet repo.

We should also add `publish = false` to prevent accidenty publishing the crate to crates.io

andygrove reviewed (approved):
LGTM. Thanks @parthchandra. I left some more comments on the Cargo toml metadata.

parthchandra commented:
> Btw I dont see the `hdfs` comet crate to refer to local `fs-hdfs`, or is it plan for follow up PR ?

@comphead Yes, there will be follow up PRs to move to this. 

@andygrove also addressed your comments.

parthchandra commented:
merged. Thank you for the review @andygrove @comphead
